server/src/processing/HeaderFooterDetectionModule/HeaderFooterDetectionModule.ts

```diff
@@ -150,25 +189,44 @@ export class HeaderFooterDetectionModule extends Module<Options> {
         `top: ${doc.margins.top}, bottom: ${doc.margins.bottom}, ` +
         `left: ${doc.margins.left}, right: ${doc.margins.right}`,
     );
+  }
 
-    doc.pages.forEach(page => {
-      const headerElements: Element[] = page.getElementsSubset(
-        new BoundingBox(0, 0, page.width, doc.margins.top),
-      );
-
-      const footerElements: Element[] = page.getElementsSubset(
-        new BoundingBox(0, doc.margins.bottom, page.width, page.height - doc.margins.bottom),
-      );
-
-      for (const element of footerElements) {
-        element.properties.isFooter = true;
-      }
-
-      for (const element of headerElements) {
-        element.properties.isHeader = true;
+  private setHeaderAndFooter(doc: Document, page: Page) {
+    const headerElements: Element[] = page.getElementsSubset(
+      new BoundingBox(0, 0, page.width, doc.margins.top),
+    );
+    const footerElements: Element[] = page.getElementsSubset(
+      new BoundingBox(0, doc.margins.bottom, page.width, page.height - doc.margins.bottom),
+    );
+    for (const element of footerElements) {
+      element.properties.isFooter = true;
+    }
+    for (const element of headerElements) {
+      element.properties.isHeader = true;
+    }
+  }
+  private countPageToTreat(docLength: number): number {
+    let nbPagesToTreat: number = docLength;
+    this.options.ignorePages.forEach(p => {
+      if (p > 0 && p <= docLength) {
+        nbPagesToTreat = nbPagesToTreat - 1;
       }
     });
-    logger.debug('Done with marginals detection.');
-    return doc;
+    return nbPagesToTreat;
+  }
+
+  private storePageSimilarSize(pagesStoredBySize: Page[][], page: Page) {
+    const indexValue = pagesStoredBySize.findIndex(
+      pageSize =>
+        pageSize[0].width * 0.9 <= page.width &&
```

Comment:
Instead of using fix ratio (0.9 and 1.1) I suggest you put this ratio in the config with a default value of 10%. This will allow user to fine tune their results.
- once add to the config, please also document the parameter in the module documentation

Reply:
Good point, it's done now.